### PR TITLE
0017395: Implement support of TGW in AWS GOV cloud

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: aviatrix/resource_aviatrix_aws_tgw.go [hunk 1]
```diff
@@ -141,6 +141,13 @@ func resourceAviatrixAWSTgw() *schema.Resource {
 					},
 				},
 			},
+			"cloud_type": {
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_aws_tgw.go:144]: Should this be `ForceNew`? If not, do we need to modify the Update function as well?

Comment (inc1t3Ful) [reply aviatrix/resource_aviatrix_aws_tgw.go:144]: If this was ForceNew, existing TGWs would be deleted; need to be mindful of backward-compatibility, though I totally understand why ForceNew would be useful here.

Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_aws_tgw.go:144]: Yes, I added a commit to block updating cloud_type.

File: aviatrix/resource_aviatrix_aws_tgw.go [hunk 2]
```diff
@@ -141,6 +141,13 @@ func resourceAviatrixAWSTgw() *schema.Resource {
...
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				Description:  "Type of cloud service provider, requires an integer value. Supported for AWS (1) and AWS GOV (256). Default value: 1.",
+				ValidateFunc: validation.IntInSlice([]int{1, 256}),
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_aws_tgw.go]:
```suggestion
				ValidateFunc: validation.IntInSlice([]int{goaviatrix.AWS, goaviatrix.AWSGOV}),
```

Could use the cloud type constants here.

**Review by CyrusJavan — APPROVED** [review]
Looks good to me.

**Pull request merged**